domain/ide/workspace/folder.go

```diff
@@ -193,17 +193,20 @@ func (f *Folder) processResults(scanData snyk.ScanData) {
 	// TODO: perform issue diffing (current <-> newly reported)
 	// Update diagnostic cache
 	for _, issue := range scanData.Issues {
-		cachedIssues, _ := f.documentDiagnosticCache.Load(issue.AffectedFilePath)
+		uniqueIssueId := f.getUniqueIssueID(issue)
+
+		cachedIssues, _ := f.documentDiagnosticCache.Load(uniqueIssueId)
+
 		if cachedIssues == nil {
 			cachedIssues = []snyk.Issue{}
 		}
 
-		if !dedupMap[f.getUniqueIssueID(issue)] {
+		if !dedupMap[uniqueIssueId] {
 			cachedIssues = append(cachedIssues, issue)
 			incrementSeverityCount(&scanData, issue)
 		}
 
-		f.documentDiagnosticCache.Store(issue.AffectedFilePath, cachedIssues)
+		f.documentDiagnosticCache.Store(uniqueIssueId, cachedIssues)
```

Comment:
**Before key fix** the cache was always empty:

![mismatch-key](https://github.com/snyk/snyk-ls/assets/1948377/17a20ab2-21f4-4267-b35c-c90a90952bb8)


```bash
[Info  - 22:46:38] createDedupMap - [DEBUG:IDE-132] Creating dedupMap with issue ID and affected file path as key issueCount=1
[Info  - 22:46:38] processResults - [DEBUG:IDE-132] Initial dedupMap size after creation dedupMapSize=0
[Info  - 22:46:38] processResults - [DEBUG:IDE-132] Checking if issue exists in dedupMap issueInDedupMap=false key=/Users/cata/git/hackathon/repository-one/hello.js uniqueIssueID=javascript/PureFunctionReturnValueIgnored|/Users/cata/git/hackathon/repository-one/hello.js
[Info  - 22:46:38] processResults - [DEBUG:IDE-132] Added issue to cachedIssues because it wasn't in dedupMap cachedIssuesAfterAdd=1 key=/Users/cata/git/hackathon/repository-one/hello.js
[Info  - 22:46:38] processResults - [DEBUG:IDE-132] Stored updated issues in document diagnostic cache cachedIssuesToStore=1 key=/Users/cata/git/hackathon/repository-one/hello.js
=== After saving the file ===
[Info  - 22:47:38] createDedupMap - [DEBUG:IDE-132] Creating dedupMap with issue ID and affected file path as key issueCount=1
[Info  - 22:47:38] processResults - [DEBUG:IDE-132] Initial dedupMap size after creation dedupMapSize=0
[Info  - 22:47:38] processResults - [DEBUG:IDE-132] Checking if issue exists in dedupMap issueInDedupMap=false key=/Users/cata/git/hackathon/repository-one/hello.js uniqueIssueID=javascript/PureFunctionReturnValueIgnored|/Users/cata/git/hackathon/repository-one/hello.js
[Info  - 22:47:38] processResults - [DEBUG:IDE-132] Added issue to cachedIssues because it wasn't in dedupMap cachedIssuesAfterAdd=1 key=/Users/cata/git/hackathon/repository-one/hello.js
[Info  - 22:47:38] processResults - [DEBUG:IDE-132] Stored updated issues in document diagnostic cache cachedIssuesToStore=1 key=/Users/cata/git/hackathon/repository-one/hello.js
```

**After key fix**, we store issues in the cache:

![fix-key](https://github.com/snyk/snyk-ls/assets/1948377/d5765142-55a9-4158-b37d-c396e5318835)



```bash
[Info  - 22:53:30] createDedupMap - [DEBUG:IDE-132] Creating dedupMap with issue ID and affected file path as key issueCount=1
[Info  - 22:53:30] processResults - [DEBUG:IDE-132] Initial dedupMap size after creation dedupMapSize=0
[Info  - 22:53:30] processResults - [DEBUG:IDE-132] Checking if issue exists in dedupMap issueInDedupMap=false key=/Users/cata/git/hackathon/repository-one/hello.js uniqueIssueID=javascript/PureFunctionReturnValueIgnored|/Users/cata/git/hackathon/repository-one/hello.js
[Info  - 22:53:30] processResults - [DEBUG:IDE-132] Added issue to cachedIssues because it wasn't in dedupMap cachedIssuesAfterAdd=1 key=/Users/cata/git/hackathon/repository-one/hello.js
[Info  - 22:53:30] processResults - [DEBUG:IDE-132] Stored updated issues in document diagnostic cache cachedIssuesToStore=1 key=/Users/cata/git/hackathon/repository-one/hello.js
=== After saving the file ===
[Info  - 22:54:07] createDedupMap - [DEBUG:IDE-132] Creating dedupMap with issue ID and affected file path as key issueCount=1
[Info  - 22:54:07] processResults - [DEBUG:IDE-132] Initial dedupMap size after creation dedupMapSize=1
[Info  - 22:54:07] processResults - [DEBUG:IDE-132] Found issues in document diagnostic cache cachedIssueCount=1 key=/Users/cata/git/hackathon/repository-one/hello.js
[Info  - 22:54:07] processResults - [DEBUG:IDE-132] Checking if issue exists in dedupMap issueInDedupMap=true key=/Users/cata/git/hackathon/repository-one/hello.js uniqueIssueID=javascript/PureFunctionReturnValueIgnored|/Users/cata/git/hackathon/repository-one/hello.js
[Info  - 22:54:07] processResults - [DEBUG:IDE-132] Skipping issue as it's already in dedupMap key=/Users/cata/git/hackathon/repository-one/hello.js uniqueIssueID=javascript/PureFunctionReturnValueIgnored|/Users/cata/git/hackathon/repository-one/hello.js
[Info  - 22:54:07] processResults - [DEBUG:IDE-132] Stored updated issues in document diagnostic cache cachedIssuesToStore=1 key=/Users/cata/git/hackathon/repository-one/hello.js
```